From 5cce79d4d705afb887d80633aec471bd083dd5f6 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@serpentos.com>
Date: Mon, 5 Jul 2021 11:08:53 +1000
Subject: [PATCH] sysdeps: Add support for default directories

In Serpent OS, we use /usr/lib for our 64bit libs and /usr/lib32 for our 32bit implementation.

Teach glibc these paths natively so it knows where to find them.

Signed-off-by: Peter O'Connor <sunnyflunk@serpentos.com>
---
 sysdeps/unix/sysv/linux/x86_64/dl-cache.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sysdeps/x86_64/dl-cache.h b/sysdeps/x86_64/dl-cache.h
index 911f6083..3a365acb 100644
--- a/sysdeps/x86_64/dl-cache.h
+++ b/sysdeps/x86_64/dl-cache.h
@@ -27,7 +27,7 @@
       size_t len = strlen (dir);				\
       char path[len + 4];					\
       memcpy (path, dir, len + 1);				\
-      if (len >= 6 && ! memcmp (path + len - 6, "/lib64", 6))	\
+      if (len >= 4 && ! memcmp (path + len - 4, "/lib", 4))	\
 	{							\
 	  len -= 2;						\
 	  path[len] = '\0';					\
@@ -39,10 +39,12 @@
 	  path[len] = '\0';					\
 	}							\
       add_dir (path);						\
-      if (len >= 4 && ! memcmp (path + len - 4, "/lib", 4))	\
+      if (len >= 6 && ! memcmp (path + len - 6, "/lib32", 6))	\
 	{							\
 	  memcpy (path + len, "64", 3);				\
 	  add_dir (path);					\
+	  memcpy (path + len, "32", 3);				\
+	  add_dir (path);					\
 	  memcpy (path + len, "x32", 4);				\
 	  add_dir (path);					\
 	}							\
-- 
2.31.1

